lib/vm.js

```diff
@@ -79,21 +79,21 @@ Script.prototype.runInNewContext = function(sandbox, options) {
   return this.runInContext(context, options);
 };
 
+function validateType(prop, propName, type = 'string') {
+  if (prop !== undefined &&
+    typeof prop !== type) {
+    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', propName,
+                               type, prop);
+  }
+}
+
 function getContextOptions(options) {
   const contextOptions = options ? {
     name: options.contextName,
     origin: options.contextOrigin
   } : {};
-  if (contextOptions.name !== undefined &&
-      typeof contextOptions.name !== 'string') {
-    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'options.contextName',
-                               'string', contextOptions.name);
-  }
-  if (contextOptions.origin !== undefined &&
-      typeof contextOptions.origin !== 'string') {
-    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'options.contextOrigin',
-                               'string', contextOptions.origin);
-  }
+  validateType(contextOptions.name, 'options.contextName');
+  validateType(contextOptions.origin, 'options.contextOrigin');
```

Comment:
Please do not change the error message itself here. So please stick to `contextOptions.name` and `contextOptions.origin`.

Reply:
@BridgeAR, the error messages are as it was.